perun-dispatcher/src/main/java/cz/metacentrum/perun/dispatcher/scheduling/TaskScheduler.java

```diff
@@ -147,6 +147,10 @@ public void run() {
 		log.debug("Pool contains {} tasks in total", schedulingPool.getSize());
 		TaskSchedule schedule;
 		while (!shouldStop()) {
+			if (tasksManagerBl.isSuspendedTasksPropagation()) {
```

Comment:
You could probably move this to the `try/catch` below and make method `waitForResumingPropagation()` to throw `InterruptedException`, so you don't have to repeat the catch implementation (similar to `getWaitingTaskSchedule()`)